sensorprocess/lidarsensorprocess.go

```diff
@@ -16,128 +16,71 @@ import (
 
 // StartLidar polls the lidar to get the next sensor reading and adds it to the cartofacade.
 // Stops when the context is Done.
-func (config *Config) StartLidar(ctx context.Context) bool {
+func (config *Config) StartLidar(ctx context.Context) {
 	for {
 		select {
 		case <-ctx.Done():
-			return false
+			return
 		default:
-			if jobDone := config.addLidarReading(ctx); jobDone {
-				config.Logger.Info("Beginning final optimization")
-				err := config.RunFinalOptimizationFunc(ctx, config.InternalTimeout)
-				if err != nil {
-					config.Logger.Error("Failed to finish processing all sensor readings: ", err)
-				}
-				return true
```

Comment:
`jobDone` is always `false` in online mode, so I could remove the hole block of code if `jobDone` were `true`.